pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/BasePatch.java

```diff
@@ -279,17 +281,24 @@ protected List<NameValue> setWorkflowRelatedFields(@NonNull Patch patch) {
                                     patch.getNameValues(), patch.getDdpParticipantId()), "sent"));
                 }
             }
-        } else if (patch.getNameValue().getName().equals("oD.unableObtainTissue") && !(boolean) patch.getNameValue().getValue()) {
-            boolean hasReceivedDate = new OncHistoryDetailDaoImpl().hasReceivedDate(getOncHistoryDetailId(patch));
+        } else if (patch.getNameValue().getName().equals("oD.unableObtainTissue")) {
+            if (!(boolean) patch.getNameValue().getValue()) {
+                boolean hasReceivedDate = new OncHistoryDetailDaoImpl().hasReceivedDate(getOncHistoryDetailId(patch));
 
-            if (hasReceivedDate) {
-                nameValues.add(setAdditionalValue("oD.request",
-                        new Patch(patch.getId(), PARTICIPANT_ID, patch.getParentId(), patch.getUser(), patch.getNameValue(),
-                                patch.getNameValues(), patch.getDdpParticipantId()), "received"));
+                if (hasReceivedDate) {
+                    nameValues.add(setAdditionalValue("oD.request",
+                            new Patch(patch.getId(), PARTICIPANT_ID, patch.getParentId(), patch.getUser(), patch.getNameValue(),
```

Comment:
Perhaps simplify this code by setting the 'additionalValue' value in a conditional and making just one complex nested call. (I appreciate some of the code was already this way but this is a great opportunity to fix sub-par code and greatly improve readability.)